tools/credentials_test.go

```diff
@@ -27,22 +29,35 @@ var _ = Describe("Credentials", func() {
 				InBar: "in-bar",
 			},
 		}
-	})
 
-	JustBeforeEach(func() {
-		var err error
 		secretData, err = tools.ToCredentialsSecretData(credsObject)
-		Expect(err).NotTo(HaveOccurred())
 	})
 
-	It("creates credentials secret data", func() {
-		Expect(secretData).To(MatchAllKeys(Keys{
-			tools.CredentialsSecretKey: MatchJSON(`{
-				"Foo": "foo",
-				"Bar": {
-					"InBar": "in-bar"
-				}
-			}`),
-		}))
+	When("ToCredentialsSecretData", func() {
+		It("successfully creates credentials secret data", func() {
+			Expect(err).NotTo(HaveOccurred())
+			Expect(secretData).To(MatchAllKeys(Keys{
+				tools.CredentialsSecretKey: MatchJSON(`{
+					"Foo": "foo",
+					"Bar": {
+						"InBar": "in-bar"
+					}
+				}`),
+			}))
+		})
+	})
+
+	When("ToDecodedSecretDataCredentials", func() {
+		BeforeEach(func() {
+			decodedCredentials, err = tools.ToDecodedSecretDataCredentials(secretData)
+			Expect(err).NotTo(HaveOccurred())
```

Comment:
The error should be checked either here, or in the `It`, no need to check it twice. Up to you to decide where.